magento/magento2
# In admin, url params may get url encoded more than once

### Description
The parent method `getUrl` should be called only once. For this purpose, I have moved a call to the `parent::getUrl()` method under the if condition, so now it's getting called only once. And as the result URL parameters no longer encoded more than once.

### Fixed Issues (if relevant)

1. magento/magento2#11478: In admin, url params may get url encoded more than once
2. magento/magento2#11343: Search term in admin grid column filter is replaced with unicode characters

### Manual testing scenarios
1. Go to Attribute Set grid in Admin Panel and try to filter by name.
2. Change sorting direction.
3. URL parameters no longer encoded multiple times and everything works correctly.

### Contribution checklist
